### PR TITLE
Wayland

### PR DESCRIPTION
**Pull request by dylanaraps**
This is for later.

Adds three sway packages:

- `sway` regular sway 
- `sway-no-seat` sway with embedded/static libseat/wlroots. (*for folks who don't want a seat daemon*)
- `sway-tiny` just the sway binary with embedded/static libseat/wlroots, and less dependencies (no pango, cairo, pcre, json-c). (*for folks who just need window management - no bar, ipc, etc*)

Adds two foot (terminal emulator) packages:

- `foot` regular foot.
- `foot-pgo` does profile guided optimization (running wayland session required).

New packages (extra):

- `pcre`
- `json-c`

New packages (wayland):

- `wlsunset`
- `wl-clipboard`
- `slurp`
- `wbg`
- `grim`
- `foot*`
- `sway*`
- `libseat`
- `wlroots`
- `wayland`
- `wayland-protocols`

Removed:

- All of xorg/
- mtdev (vendored in libinput)
- libevdev (vendored in libinput)

TODO:

- [x] firefox (*building*)
- [x] remove libevdev dependency from sway.
- [ ] firefox (fix webrtc). (will be fixed upstream regardless)

### CODE REVIEW

**Comment by konimex** [comment]
The download link for `xkeyboard-config` returns 404 for me.

**Comment by dylanaraps** [comment]
@konimex Fixed.

**Comment by dylanaraps** [comment]
Added [foot](https://codeberg.org/dnkl/foot) in two flavors.

**Comment by git-bruh** [comment]
This patch might be of interest for firefox: https://raw.githubusercontent.com/git-bruh/kiss-repo/master/wayland/firefox/patches/wayland.patch

**Comment by dylanaraps** [comment]
> This patch might be of interest for firefox: https://raw.githubusercontent.com/git-bruh/kiss-repo/master/wayland/firefox/patches/wayland.patch

Just what I needed. Thanks.

**Comment by dylanaraps** [comment]
OK. So the patch doesn't work with webrtc. Disabled webrtc for now.

File: extra/firefox/build [hunk 1]
```diff
@@ -76,6 +76,7 @@ ac_add_options --disable-dbus
 ac_add_options --disable-gold
 ac_add_options --disable-tests
 ac_add_options --disable-vtune
+ac_add_options --disable-webrtc
```
Comment (jedahan) [inline extra/firefox/build:79]: Would you mind adding a comment in the build file that `--disable-webrtc` is needed due to conflicts with `whatever.patch`, so people in the future know this is less about disabling bloat and more a limitation of firefox? (And maybe someone can take a stab at dealing with that).

Comment (dylanaraps) [reply extra/firefox/build:79]: This should be fixed upstream shortly. Will add a comment for now however. https://bugzilla.mozilla.org/show_bug.cgi?id=1661450

Thanks

**Comment by dylanaraps** [comment]
Can confirm that Firefox works.

**Comment by dylanaraps** [comment]
Apologies for the merge commits. Botched it when I went to merge. lol

**Pull request merged**